src/spec.rs

```diff
@@ -174,27 +180,22 @@ pub(crate) trait FixedSizeBlock: Pod {
     #[allow(clippy::wrong_self_convention)]
     fn from_le(self) -> Self;
 
-    #[allow(dead_code)]
-    fn interpret(input_block: &[u8]) -> ZipResult<Self> {
-        let mut block = Self::zeroed();
-        block.as_bytes_mut().copy_from_slice(input_block);
-        let block = Self::from_le(block);
+    /// Convert endianness and check the magic value.
+    #[allow(clippy::wrong_self_convention)]
+    fn validate(self) -> ZipResult<Self> {
```

Comment:
Call this function `from_le_validated` to make its combined functionality more clear, or separate out the `from_le` call and call it `with_checked_magic`.